extensions/amp-list/0.1/amp-list.js

```diff
@@ -309,21 +364,157 @@ export class AmpList extends AMP.BaseElement {
     this.container_.dispatchEvent(event);
 
     // Change height if needed.
-    this.getVsync().measure(() => {
+    return this.getVsync().measurePromise(() => {
       const scrollHeight = this.container_./*OK*/scrollHeight;
       const height = this.element./*OK*/offsetHeight;
       if (scrollHeight > height) {
-        this.attemptChangeHeight(scrollHeight).catch(() => {});
+        return this.attemptChangeHeight(scrollHeight).catch(() => {});
+      } else {
+        Promise.resolve();
+      }
+    });
+  }
+
+  /**
+   * @private
+   */
+  setLoadMore_() {
+    if (!this.loadMoreSrc_ && !this.loadMoreOverflow_) {
+      return;
+    }
+
+    const triggerOnScroll = this.element.getAttribute('load-more') === 'auto';
+    if (triggerOnScroll) {
+      this.maybeSetupLoadMoreAuto_();
+    }
+
+    this.getVsync().mutate(() => {
+      this.loadMoreOverflow_ = this.getOverflowElement();
+      if (this.loadMoreOverflow_) {
+        this.loadMoreOverflow_.classList.toggle(
+            'amp-visible', !!this.loadMoreSrc_);
+        this.loadMoreOverflow_.onclick = () => this.loadMoreCallback_();
       }
     });
+
+    if (!this.loadMoreOverflow_ && !triggerOnScroll) {
+      user().error(TAG,
+          'load-more is specified but no means of paging (overflow or ' +
+          'load-more=auto) is not available', this);
+    }
+  }
+
+  /**
+   * @private
+   */
+  loadMoreCallback_() {
+    if (!this.loadMoreSrc_) {
+      return;
+    }
+
+    if (this.loadMoreOverflow_) {
+      this.loadMoreOverflow_.onclick = null;
+    }
+
+    this.element.setAttribute('src', this.loadMoreSrc_);
+    this.loadMoreSrc_ = null;
+    this.toggleLoadMoreLoading_(true);
+
+    return this.fetchList_(/* opt_append */ true)
+        .catch(() => this.setLoadMoreFailed_())
+        .then(() => this.toggleLoadMoreLoading_(false));
+  }
+
+  /**
+   * @private
+   */
+  getLoadMoreLoadingElement_() {
+    if (this.loadMoreLoadingElement_ === undefined) {
```

Comment:
oh I see that you are using the `undefined` vs `null` for lazy querying. Honestly I don't mind this happening in `buildCallback` (if isLoadMoreEnabled) rather than doing it lazily (`buildCallback` runs during prerender mode and is meant for querying things you may need later anyway)